packages/kbn-search-connectors/lib/update_filtering_draft.ts

```diff
@@ -37,28 +32,25 @@ export const updateFilteringDraft = async (
     created_at: filteringRule.created_at ? filteringRule.created_at : now,
     updated_at: now,
   }));
-  const draft: FilteringRules = {
+
+  const draft = {
     advanced_snippet: {
       created_at: now,
       updated_at: now,
       value: parsedAdvancedSnippet,
     },
     rules: parsedFilteringRules,
-    validation: {
-      errors: [],
-      state: FilteringValidationState.EDITED,
-    },
   };
-  const connector = await fetchConnectorById(client, connectorId);
-  if (!connector) {
-    throw new Error(`Could not find connector with id ${connectorId}`);
-  }
 
-  const result = await client.update<Connector>({
-    doc: { ...connector, filtering: [{ ...connector.filtering[0], draft }] },
-    id: connectorId,
-    index: CONNECTORS_INDEX,
+  const updateDraftFilteringResult = await client.transport.request<{ result: Result }>({
+    method: 'PUT',
+    path: `/_connector/${connectorId}/_filtering`,
+    body: draft,
   });
 
-  return result.result === 'updated' ? draft : undefined;
+  if (updateDraftFilteringResult.result === 'updated') {
+    const connector = await fetchConnectorById(client, connectorId);
+    return connector?.filtering[0].draft;
```

Comment:
```suggestion
    return connector?.filtering?.[0]?.draft;
```

Reply:
See comment above, thank you for suggestion